src/pip/_internal/download.py

```diff
@@ -46,11 +48,22 @@
 from pip._internal.utils.ui import DownloadProgressProvider
 from pip._internal.vcs import vcs
 
+# We don't try to import OpenSSL on Windows since that might result in it
+# loading C libraries, which can then not be uninstalled.
+if not WINDOWS:
```

Comment:
It might be a good idea to not duplicate this logic, and instead use the variables that urllib3 sets, see:

https://github.com/shazow/urllib3/blob/master/urllib3/contrib/pyopenssl.py#L109-L118

Reply:
orig_util_SSLContext is never None -- https://github.com/shazow/urllib3/blob/master/urllib3/util/ssl_.py#L90

I do think calling `_validate_dependencies_met` is a good idea.